httpclient5/src/main/java/org/apache/hc/client5/http/impl/async/AsyncRedirectExec.java

```diff
@@ -187,6 +192,19 @@ public AsyncDataConsumer handleResponse(
                 return asyncExecCallback.handleResponse(response, entityDetails);
             }
 
+            @Override
+            public void handleInformationResponse(final HttpResponse response) throws HttpException, IOException {
+                redirect(response);
```

Comment:
@kirill-usov Are you sure about doing a redirect on an informational response? This does not look right to me.